packages/runtime/container-runtime/src/compatUtils.ts

```diff
@@ -0,0 +1,183 @@
+ * Therefore, if the customer does not provide a compatibility mode, we will default to use N-1.
+ *
+ * However, this is not consistent with today's behavior. Some options (i.e. batching, compression) are enabled by default
+ * despite not being compatible with 1.x clients. Since the policy was introduced during 2.x's lifespan, N/N-1 compatibility
+ * by **default** will be in effect starting with 3.0. Importantly though, N/N-1 compatibility is still guaranteed with the
+ * proper configurations set.
+ *
+ * TODO: This should be updated to "2.0.0" when 3.0 is released.
+ */
+export const defaultCompatibilityMode = "pre-3.0-default";
+
+/**
+ * Available compression algorithms for op compression.
+ * @legacy
+ * @alpha
+ */
+export enum CompressionAlgorithms {
+	lz4 = "lz4",
+}
+
+/**
+ * @legacy
+ * @alpha
+ */
+export const disabledCompressionConfig: ICompressionRuntimeOptions = {
+	minimumBatchSizeInBytes: Number.POSITIVE_INFINITY,
+	compressionAlgorithm: CompressionAlgorithms.lz4,
+};
+
+export const enabledCompressionConfig = {
+	// Batches with content size exceeding this value will be compressed
+	minimumBatchSizeInBytes: 614400,
+	compressionAlgorithm: CompressionAlgorithms.lz4,
+};
+
+/**
+ * Subset of the IContainerRuntimeOptionsInternal properties which are version-dependent.
+ *
+ * @remarks
+ * When a new option is added to IContainerRuntimeOptionsInternal, we must consider if it's a version-dependent option.
+ * If it's considered version-dependent, then a corresponding entry must be added to `versionDependentOptionConfigs`. If not, then
+ * it must be omitted from this type.
+ *
+ * See [TODO: DOC LINK] for more details.
+ */
+export type IContainerRuntimeOptionsVersionDependent = Required<
+	Omit<
+		IContainerRuntimeOptionsInternal,
+		| "chunkSizeInBytes"
+		| "maxBatchSizeInBytes"
+		| "loadSequenceNumberVerification"
+		| "summaryOptions"
+		| "compatibilityMode"
+	>
+>;
+
+/**
+ * Map of version-dependent IContainerRuntimeOptionsInternal keys to their compatibility related information.
+ * The key is the option name, and the value is an object containing:
+ * - minVersionRequired: The minimum version of the container runtime that is required to use the version-dependent option
+ * - legacyConfig: The default config of the option when the runtime does not meets the min version requirement
+ * - modernConfig: The default config of the option when the runtime meets the min version requirement
+ *
+ * TODO: Double check with feature owners that all minVersionRequired are correct
+ */
+const versionDependentOptionConfigMap: {
+	[K in keyof IContainerRuntimeOptionsVersionDependent]: {
+		minVersionRequired: string;
+		legacyConfig: IContainerRuntimeOptionsVersionDependent[K];
+		modernConfig: IContainerRuntimeOptionsVersionDependent[K];
+	};
+} = {
+	enableGroupedBatching: {
+		minVersionRequired: "2.0.0-internal.4.1.0",
```

Comment:
Although the internal versions might technically be the earliest opportunity when the feature could work, IMO we should be more strict and just say 2.0.0.

Sticking to only vanilla-looking semvers here probably simplifies our life later for documenting/explaining, and/or if we want these to be easily parsable/comparable, etc.  Anyone without the historical knowledge of the pre-2.0 release era won't know what to make of these version numbers, which will be a maintainability pain point.